### DISCUSSION TITLE
Fyrox editor general installation error

### DISCUSSION THREAD

**Discussion by b0id** [FyroxEngine/Fyrox#395] [Q&A]
Hi,

### I am most of the way through building the fyrox editor and after going through the various depencies one by one, I have reached a point where I can't find a solution.  

`user@terminal;~/dir $   cargo run --package editor --release 
warning: field "scene" is never read
  --> game/src/lib.rs:31:5
   |
30 pub struct Game {
   |            ---- field in this struct
31 |     scene: Handle<Scene>,
   |     ^^^^^
   |
   = note: ``#[warn(dead_code)]`` on by default`

`warning: ``my_game`` (lib) generated 1 warning
    Finished release [optimized] target(s) in 0.44s
     `Running` `target/release/editor`
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Custom("CreationErrors([OpenGlVersionNotSupported, NoAvailablePixelFormat])")', /home/ubuntustudio/.cargo/registry/src/github.com-1ecc6299db9ec823/fyroxed_base-0.15.0/src/lib.rs:506:10
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace`
  
### This is output that may be helpful.

`~/repos/rust-projects/Qbert/my_game$ glxinfo | grep "version"
server glx version string: 1.4
client glx version string: 1.4
GLX version: 1.4
    Max core profile version: 0.0
    Max compat profile version: 2.1
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 2.0
OpenGL version string: 2.1 Mesa 22.0.5
OpenGL shading language version string: 1.20
OpenGL ES profile version string: OpenGL ES 2.0 Mesa 22.0.5
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 1.0.16`

If one of you don't mind pointing me in the right direction here, I would appreciate it.

**Comment by mrDIMAS** [comment]
Hi! You need a GPU with at least OpenGL 3.3 Core support. As I can see, your GPU supports only OpenGL 2.1. Are you running the engine on a virtual machine or your video drivers are way too outdated? What's your GPU btw?

**Comment by b0id** [comment]
Yeah its a VM w 3 cores. (and probably not enough VRAM ever now that I think about it.) That is what I suspected.  Upgrading the CPU this week. NVIDIA RTX2060.  Once I get it upgraded the process for upgrading to 3.3 is straightforward? (edit:I Looked it up, looks like I am mostly shit out of luck in the matter) I had some trouble with finding more info on the OpenGL protocols. I appreciate the feedback.

**Reply by mrDIMAS** [reply]
Why not just run it without a VM? Pretty much all VM has bad support of graphics API.

**Reply by b0id** [reply]
Yeah I will have to swap over to the other partition, but currently my machine has 6 monitors across 2GPUs, and I havent hacked them together on the Linux side of things.  Thanks for letting me know.  I am curious where in the documentation it states that the required version.  I looked around, but didn't see it.  Just want to be knowledgable as much as possible.  Thanks again. Consider this issue closed.
